### PR TITLE
tests: split test_ingest_sst into two tests

### LINKED ISSUE #7266: import::sst_service::test_ingest_sst failed
import::sst_service::test_ingest_sst

Latest failed builds: 
https://internal.pingcap.net/idc-jenkins/job/tikv_ghpr_test/19498/consoleFull

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve?

Issue Number: close #7266

Problem Summary:

The test `import::sst_service::test_ingest_sst` reuses the SST ingest name and rely on them being cleaned up in time, which is not necessarily true because of concurrency.

### What is changed and how it works?

1. Split `test_ingest_sst_without_crc32` off from `test_ingest_sst`, so that the SST file names no longer collide.
2. Removed the test which asserts the SST is cleaned up at the end of `test_ingest_sst_without_crc32`. This is not a property we need to test.
3. Added a failpoint to ensure the tests won't fail even when the SST is not promptly deleted.

What's Changed:

### Check List 

Tests 

- Unit test

Side effects

### Release note

### CODE REVIEW

**Comment by BusyJay** [comment]
For now failpoint tests need to be defined in failpoint directory.

File: tests/integrations/import/sst_service.rs [hunk 1]
```diff
@@ -101,28 +101,41 @@ fn test_ingest_sst() {
...
     assert!(send_upload_sst(&import, &meta, &data).is_err());
 
     ingest.set_sst(meta.clone());
     let resp = import.ingest(&ingest).unwrap();
     assert!(!resp.has_error());
+}
+
+#[test]
+fn test_ingest_sst_without_crc32() {
+    fail::cfg("raft_before_delete_sst", "sleep(1000)").unwrap();
```
Comment (Little-Wallace) [inline tests/integrations/import/sst_service.rs:114]: If you want to make a `fail_point` wait a while for some event or signal, you can set a callback in test code and make it wait for notifying, which is a  new feature of `fail_point`. See more details in https://github.com/tikv/fail-rs/pull/50/files

**Comment by kennytm** [comment]
@BusyJay maybe i should just remove the failpoint test for simplicity? this is not the essential part of the fix (just to verify that the spurious error won't happen even with extreme delay).

**Comment by BusyJay** [comment]
You mean just remove the failpoint but keep both test cases, right? Then I'm OK with the change.

**Comment by kennytm** [comment]
/run-integration-copr-test

---

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tikv_ghpr_integration-copr-test/detail/tikv_ghpr_integration-copr-test/2454/pipeline

```
[2020-04-14T16:16:41.045Z] Statement: #507 -  SELECT COALESCE( `col_tinyint_unsigned`, ( COALESCE( ( LEAST( `col_double_unsigned_key`, 4556, ( '2000-01-09' >= ( STRCMP( `col_time_key`, `col_smallint_unsigned_key` ) ) ) ) ), ( ( STRCMP( ( `col_float_unsigned` IS NOT TRUE ), `col_char_255_key` ) ) > ( 'nmo' != -15482 ) ), '1990-06-16 17:22:56.005534' ) ), '2007-07-03', `col_float`, ( `col_float` BETWEEN 0 AND ( ( GREATEST( ( ( `col_set` <=> '2027-11-11' ) <> ( ( ( COALESCE( 'moxqnhbnkyxksjwoaajpbxxobggqewsbvtlqqjkkakmuskosyzsuhdlvfrhobgixtbeqjisgazsdqtccshcxarzvuxsjteeyxmzfxpu', 'p' ) ) IS NULL ) < ( STRCMP( ( `col_int_key` >= 'oxqnhbnkyxksjwoaajpbxxobggqewsbvtlqqjkkakmuskosyzsuhdlvfrhobgixtbeqjisgazsdqtccshcxarzvuxsjteeyxmzfxpuenwuwdczcyaumvpzhxuzftshcwrycsohwtdkrljskkbbbpvsxhzmlpqxozpcrpvqevuvdosgxtlunuhjyomjbucywsqgvbwromzotrdldqfpqvjaxfyhndbzyuuupvchaxe' ), ( -24369 != 'xqnhbnkyxksjwoaajpbxxobggqewsbvtlqqjkkakmuskosyzsuhdlvfrhobgixtbeqjisgazsdqtccshcxarzvuxsjteeyxmzfxpuenwuwdczcyaumvpzhxuzftshcwrycsohwtdkrljskkbbbpvsxhzmlpqxozpcrpvqevuvdosgxtlunuhjyomjbucywsqgvbwromz' ) ) ) ) ), '20:14:46.019164' ) ) <> 'j' ) ) ) AS field1, ISNULL( ( ( ( STRCMP( ( `col_year` IS FALSE ), 7334674943126274048 ) ) IN ( -607985949695016960, `col_varbinary_32`, '2023-03-02 19:05:21.008216', ( 0 < 0 ), `col_set_key` ) ) NOT BETWEEN ( ( 51 IS NOT NULL ) NOT IN ( NULL, ( ISNULL( ( `col_varbinary_32_key` IS UNKNOWN ) ) ) ) ) AND `col_binary_8_key` ) ) AS field2, LEAST( `col_varbinary_32_key`, `col_text_key` ) AS field3 FROM `table1000_int_autoinc` WHERE ( 0 <> `col_decimal_unsigned_key` ) IN ( ( COALESCE( ( `col_double_unsigned` = `col_char_2` ), `col_datetime_key` ) ), ( INTERVAL( ( COALESCE( '04:24:43.033324' ) ), `col_text` ) ) ) ORDER BY field1, field2, field3 LIMIT 7 /* QNO 509 CON_ID 196 */ ;
[2020-04-14T16:16:41.045Z] NoPushDown Output: 
[2020-04-14T16:16:41.045Z] field1	field2	field3
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 
[2020-04-14T16:16:41.045Z] 
[2020-04-14T16:16:41.045Z] WithPushDown Output: 
[2020-04-14T16:16:41.045Z] field1	field2	field3
[2020-04-14T16:16:41.045Z] 0	0	199229440
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	NULL
[2020-04-14T16:16:41.045Z] 0	0	1622605824
[2020-04-14T16:16:41.045Z] 0	0	NULL
```

**Comment by BusyJay** [comment]
@kennytm Does it need to be picked to release-4.0?

**Comment by kennytm** [comment]
Yes

**Pull request merged**